src/components/DataSearch.jsx

```diff
@@ -0,0 +1,380 @@
+import React, { Component } from 'react';
+import {
+  bool,
+  string,
+  object,
+  number,
+  func,
+  dataField,
+  position,
+  suggestions,
+  highlightField,
+  queryFormat,
+  fuzziness,
+  title,
+  any,
+  react,
+  themePreset
+} from '../utils/types';
+import Input, {
+  suggestionsContainer,
+  suggestions as suggestionsCss
+} from './Input';
+import Container from '../styles/Container';
+import Title from '../styles/Title';
+import {
+  deepGet,
+  getClassName,
+  getComponent,
+  hasCustomRenderer
+} from '../utils/helper';
+import Downshift from 'downshift';
+import Icons from './Icons';
+import Loader from './Loader';
+import Error from './Error';
+import SuggestionItem from '../addons/SuggestionItem';
+import NoSuggestions from './NoSuggestions';
+import Searchbase from '@appbaseio/searchbase';
+
+class DataSearch extends Component {
+  constructor(props) {
+    super(props);
+    const { index, url, dataField, credentials } = props;
+    const currentValue = props.value || props.defaultValue || '';
+
+    this.state = {
+      currentValue,
+      suggestionsList: [],
+      isOpen: false
+    };
+
+    this.searchBase = new Searchbase({
+      index,
+      url,
+      dataField,
+      credentials
+    });
+
+    this.searchBase.subscribeToStateChanges(
+      ({ suggestions }) => {
+        this.setState({
```

Comment:
@bietkul when I compare `suggestions.prev.data` and `suggestions.next.data`, they are same, seems to be a bug in `SearchBase`